### ISSUE TITLE
Initializing a Vec<QPdf> with QPdf::empty() results in all elements of the Vec pointing to the same handler

### ISSUE THREAD

**Issue by protochron** [ancwrd1/qpdf-rs#5]
Something about this code isn't working well with the way that `QPdf::empty()` is implemented:

```rust
let pdfs = vec![QPdf::empty(); number_to_init as usize];
```
What _does_ work is initializing each element of the Vec using separate QPdf::empty() calls:

```rust
let mut pdfs: Vec<QPdf> = Vec::new();
for _ in 0..length
  pdfs.push(QPdf::empty());
}
```

Using `rust-lldb` I was able to verify that initializing a Vec with `QPdf::empty()` set all elements of the Vec to use the same pointer in the `Handler`, or specifically whatever is pointed to by `qpdf_sys::qpdf_init()`

**Comment by ancwrd1** [comment]
Yeah, the `vec!` macro expects that the element has a `Clone` implementation. The QPdf structure has a reference-counted internal handle and a cheap "clone" operation so it can be easily passed over as a function parameter, pointing to the same data. This was originally by design, which unfortunately has this side-effect when constructing a Vec.
I am not sure how to resolve it easily.
